### PR TITLE
Add MPICH_RDMA_ENABLED_CUDA test

### PR DESCRIPTION
**Pull request by jgphpc**
This test will help detect issue like https://jira.cscs.ch/browse/SD-55944

### CODE REVIEW

**Comment by jgphpc** [comment]
need to add a skip call

File: checks/prgenv/cuda/cuda_aware_mpi.py [hunk 1]
```diff
@@ -138,3 +139,73 @@ def set_num_tasks(self):
...
+@rfm.simple_test
+class gpu_direct(rfm.RegressionTest):
+    descr = 'gpuDirect MPI test'
+    sourcepath = 'gpudirect.cpp'
+    valid_systems = ['daint:gpu', 'dom:gpu']
+    valid_prog_environs = ['PrgEnv-cray']
+    build_system = 'SingleSource'
+    maintainers = ['JG']
+    tags = {'production', 'scs'}
+    valid_prog_environs = ['PrgEnv-cray']
```
Comment (teojgo) [inline checks/prgenv/cuda/cuda_aware_mpi.py:153]: delete this

File: checks/prgenv/cuda/cuda_aware_mpi.py [hunk 2]
```diff
@@ -138,3 +139,73 @@ def set_num_tasks(self):
...
+                     f'{s} not in {self.valid_systems}')
+        self.skip_if(e not in self.valid_prog_environs,
+                     f'{e} not in {self.valid_prog_environs}')
+        bad_cdt = ['cdt/21.09', 'cdt/20.08']
+        self.skip_if(m in bad_cdt, f'{m} in {bad_cdt}')
+        self.modules = ['cudatoolkit', m]
+
+    @run_before('compile')
+    def set_compilers(self):
+        self.build_system.options = [
```
Comment (teojgo) [inline checks/prgenv/cuda/cuda_aware_mpi.py:172]: This should be set via the cxxflags and the ldflags

File: checks/prgenv/cuda/cuda_aware_mpi.py [hunk 3]
```diff
@@ -138,3 +139,73 @@ def set_num_tasks(self):
...
+            '-I/usr/local/cuda/targets/x86_64-linux/include',
+            '-L/usr/local/cuda/targets/x86_64-linux/lib',
+            '-lcudart',
+        ]
+
+    @run_before('run')
+    def set_env(self):
```
Comment (teojgo) [inline checks/prgenv/cuda/cuda_aware_mpi.py:179]: All this can go to the class body

File: checks/prgenv/cuda/cuda_aware_mpi.py [hunk 4]
```diff
@@ -138,3 +139,73 @@ def set_num_tasks(self):
...
+        }
+        self.prerun_cmds = [
+            'export LD_LIBRARY_PATH=$CRAY_LD_LIBRARY_PATH:$LD_LIBRARY_PATH',
+            # test will fail when built with cdt<=21.09 (cray-mpich<=7.7.18):
+            # PASS:rk=1     // 1st call
+            # FAIL:rk=1:-1  //  2d call
+        ]
+
+    @run_before('run')
+    def set_num_tasks(self):
```
Comment (teojgo) [inline checks/prgenv/cuda/cuda_aware_mpi.py:193]: Move these to the class body

File: checks/prgenv/cuda/cuda_aware_mpi.py [hunk 5]
```diff
@@ -138,3 +139,73 @@ def set_num_tasks(self):
...
+        self.num_tasks = 2
+        self.num_tasks_per_node = 1
+
+    @run_before('sanity')
```
Comment (teojgo) [inline checks/prgenv/cuda/cuda_aware_mpi.py:197]: Use the new syntax for the sanity patterns

**Review by teojgo — CHANGES_REQUESTED** [review]

File: checks/prgenv/cuda/cuda_aware_mpi.py (hunk 5 above)
Comment (jgphpc) [reply checks/prgenv/cuda/cuda_aware_mpi.py:165]: ?

Comment (vkarak) [reply checks/prgenv/cuda/cuda_aware_mpi.py:165]:
```suggestion
    @sanity_function
    def assert_sanity(self):
        result = sn.count(sn.extractall(r'^(PASS):', self.stdout, 1))
        return sn.assert_reference(result, 2, 0, 0)
```

**Pull request closed without merging**